packages/babel-core/src/gensync-utils/forward.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import gensync, { type Handler } from "gensync";
+
+type MaybePromise<T> = T | Promise<T>;
+
+export default function forward<ActionArgs: mixed[], ActionReturn, Return>(
+  action: (...args: ActionArgs) => Handler<ActionReturn>,
+  cb: (
+    adapted: (...args: ActionArgs) => MaybePromise<ActionReturn>,
+  ) => MaybePromise<Return>,
+): Handler<Return> {
+  return isAsync(kind => {
+    const adapted = cachedGensync<ActionArgs, ActionReturn>(action)[kind];
+    return cb(adapted);
+  });
+}
+
+export const isAsync = gensync<[any], any>({
```

Comment:
Is there a way we can avoid `any`? I tried really hard to never use `any` in the core codebase, which is why I keep mentioning it.

Reply:
This can definitely be removed, I don't know why I used it